systemtest/src/test/java/io/strimzi/systemtest/MultipleNamespaceST.java

```diff
@@ -0,0 +1,180 @@
+
+        LOGGER.info("Waiting for creation {} in namespace {}", kafkaName, SECOND_NAMESPACE);
+        kubeClient.namespace(SECOND_NAMESPACE);
+        kubeClient.waitForStatefulSet(kafkaName, 3);
+    }
+
+
+    /**
+     * Test the case when MirrorMaker will be deployed in different namespace across multiple namespaces
+     */
+    @Test
+    @Tag(REGRESSION)
+    void testDeployMirrorMakerAcrossMultipleNamespace() throws InterruptedException {
+        String kafkaName = CLUSTER_NAME + "-target";
+        String kafkaSourceName = kafkaClusterName(CLUSTER_NAME);
+        String kafkaTargetName = kafkaClusterName(kafkaName);
+
+        secondNamespaceResources.kafkaEphemeral(kafkaName, 3).done();
+        secondNamespaceResources.kafkaMirrorMaker(CLUSTER_NAME, kafkaSourceName, kafkaTargetName, "my-group", 1, false).done();
+
+        LOGGER.info("Waiting for creation {} in namespace {}", CLUSTER_NAME + "-mirror-maker", SECOND_NAMESPACE);
+        kubeClient.namespace(SECOND_NAMESPACE);
+        kubeClient.waitForDeployment(CLUSTER_NAME + "-mirror-maker", 1);
+    }
+
+
+
+    @BeforeAll
+    static void createClassResources(TestInfo testInfo) {
+        LOGGER.info("Creating resources before the test class");
+        classResources = new Resources(namespacedClient());
+        classResources().kafkaEphemeral(CLUSTER_NAME, 3)
+            .editSpec()
+                .editEntityOperator()
+                    .editTopicOperator()
+                        .withWatchedNamespace(SECOND_NAMESPACE)
+                    .endTopicOperator()
+                .endEntityOperator()
+            .endSpec()
+            .done();
+
+        testClass = testInfo.getTestClass().get().getSimpleName();
+    }
+
+    @AfterAll
+    static void deleteClassResources() {
+        LOGGER.info("Deleting resources after the test class");
+        classResources.deleteResources();
+        classResources = null;
+    }
+
+    @BeforeEach
+    void createSecondNamespaceResources() {
+        kubeClient.namespace(SECOND_NAMESPACE);
+        secondNamespaceResources = new Resources(namespacedClient());
+        kubeClient.namespace(DEFAULT_NAMESPACE);
+    }
+
+    @AfterEach
+    void deleteSecondNamespaceResources() {
+        secondNamespaceResources.deleteResources();
+        kubeClient.namespace(DEFAULT_NAMESPACE);
+    }
+
+    private static Resources classResources() {
+        return classResources;
+    }
+
+    private void deployNewTopic() {
+        LOGGER.info("Creating topic {} in namespace {}", TOPIC, SECOND_NAMESPACE);
+        kubeClient.namespace(SECOND_NAMESPACE);
+        kubeClient.create(new File(TOPIC_INSTALL_DIR));
+        TestUtils.waitFor("wait for 'my-topic' to be created in Kafka", 120000, 5000, () -> {
+            kubeClient.namespace(DEFAULT_NAMESPACE);
+            List<String> topics2 = listTopicsUsingPodCLI(CLUSTER_NAME, 0);
+            return topics2.contains(TOPIC);
+        });
+    }
+
+    private void deleteNewTopic() {
```

Comment:
Same comment about the namespace this is implicitly using: Include it in the method name, or pass it as an argument.

Reply:
Namespace and topic name added as an argument.